### PR TITLE
WIP: Features/group offer categories

### PR DESCRIPTION
**Pull request by nuritnt**
# [Story in Trello](https://trello.com/c/iM0dsubq/3-3-als-superadmin-m%C3%B6chte-ich-eine-liste-von-gruppenangebots-kategorie-verwalten)

### Done for approval?

* [ ] Acceptance criteria are met
* [ ] Code quality checks are green
* [ ] Readme updated if needed
* [ ] Story under test
* [ ] Mobile works
* [ ] Seeds created
* [ ] Translated
* [ ] Code is reviewed

### Done after the merge?

* [ ] Deployed to staging after the merge
* [ ] Tested manually on staging

### CODE REVIEW

File: app/policies/group_offer_category_policy.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class GroupOfferCategoryPolicy < ApplicationPolicy
+  class Scope < ApplicationScope
+    def resolve
+      all if superadmin?
+    end
+  end
+
+  alias_method :index?,                 :superadmin?
+  alias_method :show?,                  :superadmin?
+  alias_method :new?,                   :superadmin?
```
Comment (chrysalag) [inline app/policies/group_offer_category_policy.rb]: Should categories be restricted only to superadmins?

File: app/views/group_offer_categories/_form.html.slim [hunk 2]
```diff
@@ -0,0 +1,13 @@
+= simple_form_for(@group_offer_category) do |f|
+  = simple_error_notice f
+  - if policy(GroupOffer).supervisor_privileges?
```
Comment (chrysalag) [inline app/views/group_offer_categories/_form.html.slim]: Since categories are restricted to superadmins, this should not be needed. Also, the restrictions should be based on category policies and not group offer ones.

File: app/views/group_offers/index.html.slim [hunk 3]
```diff
@@ -1,6 +1,12 @@
 h1= t_title
 
-= form_navigation_btn :new
+nav.navbar.section-navigation
+  ul.list-inline
+    li= button_link t_title(:new), new_group_offer_path, 'success', dimension: 'sm'
+    - if current_user.superadmin?
```
Comment (chrysalag) [inline app/views/group_offers/index.html.slim]: this could also be checked through a policy.

File: config/routes.rb [hunk 4]
```diff
@@ -30,6 +30,8 @@
   end
   resources :reminders, only: [:index, :update, :destroy]
   resources :group_offers
+  resources :group_offer_categories, except: [:destroy]
+
```
Comment (chrysalag) [inline config/routes.rb]: NIT: Extra empty line not needed.

File: test/system/group_offer_categories_test.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+require 'application_system_test_case'
+
+class GroupOfferCategoriesTest < ApplicationSystemTestCase
+  def setup
+    @user = create :user
+    login_as @user
+  end
+
+
+  test 'create new group offer category and update it' do
```
Comment (chrysalag) [inline test/system/group_offer_categories_test.rb]: In general, each test should test only one thing. So please slip that test in two.

File: app/views/group_offers/_form.html.slim [hunk 6]
```diff
@@ -27,6 +27,8 @@
       = f.input :necessary_volunteers, collection: 1..10
     .col-xs-12.col-md-4
       = f.input :volunteer_responsible_state, collection: GroupOffer::VOLUNTEER_RESPONSIBLE_STATES
+    .col-xs-12.col-md-4
```
Comment (chrysalag) [inline app/views/group_offers/_form.html.slim]: Maybe the category selection fits better either as a separate row below the title or in the same row with the title.

File: app/views/group_offers/index.html.slim [hunk 7]
```diff
@@ -1,6 +1,12 @@
 h1= t_title
 
-= form_navigation_btn :new
+nav.navbar.section-navigation
+  ul.list-inline
+    li= button_link t_title(:new), new_group_offer_path, 'success', dimension: 'sm'
```
Comment (chrysalag) [inline app/views/group_offers/index.html.slim]: Since this button is now `sm`, I think it would look better if the new offer button at the end of the page was also small.

File: test/system/group_offer_categories_test.rb (hunk 5 above)
Comment (toupeira) [reply test/system/group_offer_categories_test.rb]: This is true for unit tests but for system tests it's sometimes easier to test a whole workflow, though it can also make it trickier to update the test in the future.

In general I can recommend writing more unit tests where you test all edge cases (they're simpler to write and run a lot faster), and using system tests for general workflows and things you can't easily test directly (e.g. JS interactions).

File: app/views/group_offer_categories/edit.html.slim [hunk 8]
```diff
@@ -0,0 +1,7 @@
+.row
```
Comment (casaper) [inline app/views/group_offer_categories/edit.html.slim]: No need to enclose block elements on full width within row and column

File: app/views/group_offer_categories/index.html.slim [hunk 9]
```diff
@@ -0,0 +1,21 @@
+.row
+  .col-xs-12
+    h1= t_attr(:group_offer_categories)
```
Comment (casaper) [inline app/views/group_offer_categories/index.html.slim]: No need to enclose block elements on full width within row and column

File: app/views/group_offer_categories/index.html.slim [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+= form_navigation_btn :new
+
+.row
```
Comment (casaper) [inline app/views/group_offer_categories/index.html.slim]: If this table should be enclosed in a div it should be `.table-responsive` and not row.

File: app/views/group_offer_categories/new.html.slim [hunk 11]
```diff
@@ -0,0 +1,7 @@
+.row
+  .col-xs-12
+    h1= t_title(:new)
```
Comment (casaper) [inline app/views/group_offer_categories/new.html.slim]: No need to enclose block elements on full width within row and column

File: app/views/group_offer_categories/show.html.slim [hunk 12]
```diff
@@ -0,0 +1,18 @@
+.row
+    .col-xs-12
+      h1= t_attr(:group_offer_category)
```
Comment (casaper) [inline app/views/group_offer_categories/show.html.slim]: No need to enclose block elements on full width within row and column

File: app/views/group_offer_categories/show.html.slim [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+.row
+  .col-xs-12
```
Comment (casaper) [inline app/views/group_offer_categories/show.html.slim]: enclosing this table in a row is of no use

File: app/views/group_offers/index.html.slim [hunk 14]
```diff
@@ -1,6 +1,12 @@
...
+    - if current_user.superadmin?
+      li= button_link t('group_offer_categories'), group_offer_categories_path, dimension: 'sm'
+
+
 
 .row
   .col-xs-12
```
Comment (casaper) [inline app/views/group_offers/index.html.slim]: Dont put tables within row and column divs

File: config/locales/de.yml [hunk 15]
```diff
@@ -79,6 +79,9 @@ de:
       full: Voll
       open: Offen
       partially_occupied: Teilweise besetzt
+    category_states: &id-category_states
+      active:
```
Comment (chrysalag) [inline config/locales/de.yml]: translation missing
